lib/puppet/functions/azure_key_vault/lookup.rb

```diff
@@ -3,14 +3,15 @@
 Puppet::Functions.create_function(:'azure_key_vault::lookup') do
   dispatch :lookup_key do
     param 'Variant[String, Numeric]', :secret_name
-    param 'Struct[{vault_name => String, vault_api_version => String, metadata_api_version => String}]', :options
+    param 'Struct[{vault_name => String, vault_api_version => String, metadata_api_version => String, key_replacement_token => String}]', :options
     param 'Puppet::LookupContext', :context
   end
 
   def lookup_key(secret_name, options, context)
     # This is a reserved key name in hiera
     return context.not_found if secret_name == 'lookup_options'
-    return context.cached_value(secret_name) if context.cache_has_key(secret_name)
+    keyvault_object_name = TragicCode::Azure.normalize_object_name(secret_name, options['key_replacement_token'] || '-')
```

Comment:
what do you think about renaming the function and variable name to this?

```
normalized_secret_name = TragicCode::Azure.normalize_secret_name(..)
```

Reply:
The normalization logic should applye equally well for KeyVault Keys and Certificates as well.

I'm not sure whether they'd be used for automatic parameter lookup, but I can see #54 discussing the use of other object types, so maybe leave it as is in case there's use there?

But I can see the wisdom in renaming the local var, so I've split the difference and done that.

Reply:
LGTM